core/src/main/java/org/axonframework/queryhandling/responsetypes/ResponseType.java

```diff
@@ -0,0 +1,46 @@
+package org.axonframework.queryhandling.responsetypes;
+
+import java.io.Serializable;
+import java.lang.reflect.Type;
+
+/**
+ * Specifies the expected response type required when performing a query through the
+ * {@link org.axonframework.queryhandling.QueryBus}/{@link org.axonframework.queryhandling.QueryGateway}.
+ * By wrapping the response type as a generic {@code R}, we can easily service the expected response as a single
+ * instance, a list, a page etc., based on the selected implementation even while the query handler return type might be
+ * slightly different.
+ * </p>
+ * It is in charge of matching the response type of a query handler with the given generic {@code R}.
+ * If this match returns true, it signals the found query handler can handle the intended query.
+ * As a follow up, the response retrieved from a query handler should move through the
+ * {@link ResponseType#convert(Object)} function to guarantee the right response type is returned.
+ *
+ * @param <R> the generic type of this {@link ResponseType} to be matched and converted.
+ * @author Steven van Beelen
+ * @since 3.2
+ */
```

Comment:
Would be nice to have a javadoc reference (`@see`) to `ResponseTypes`, just in case some stumbles on this class, looking for static creator methods.

Reply:
Nice suggestion, added a `@see ResponseTypes` to the javadoc here.